homeassistant/components/nextcloud/sensor.py

```diff
@@ -0,0 +1,190 @@
+    "storage_num_shares_user": ["nextcloud", "shares", "num_shares_user"],
+    "storage_num_shares_groups": ["nextcloud", "shares", "num_shares_groups"],
+    "storage_num_shares_link": ["nextcloud", "shares", "num_shares_link"],
+    "storage_num_shares_mail": ["nextcloud", "shares", "num_shares_mail"],
+    "storage_num_shares_room": ["nextcloud", "shares", "num_shares_room"],
+    "storage_num_shares_link_no_password": [
+        "nextcloud",
+        "shares",
+        "num_shares_link_no_password",
+    ],
+    "storage_num_fed_shares_sent": ["nextcloud", "shares", "num_fed_shares_sent"],
+    "storage_num_fed_shares_received": [
+        "nextcloud",
+        "shares",
+        "num_fed_shares_received",
+    ],
+    "server_webserver": ["server", "webserver"],
+    "server_php_version": ["server", "php", "version"],
+    "server_php_memory_limit": ["server", "php", "memory_limit"],
+    "server_php_max_execution_time": ["server", "php", "max_execution_time"],
+    "server_php_max_file_size": ["server", "php", "upload_max_filesize"],
+    "server_database_type": ["server", "database", "type"],
+    "server_database_version": ["server", "database", "version"],
+    "server_database_size": ["server", "database", "size"],
+    "active_users_last_5_minutes": ["activeUsers", "last5minutes"],
+    "active_users_last_1_hour": ["activeUsers", "last1hour"],
+    "active_users_last_24_hours": ["activeUsers", "last24hours"],
+}
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None) -> None:
+    """Set up the Nextcloud sensor."""
+    try:
+        # Fetch Nextcloud Monitor api data
+        ncm = NextcloudMonitor(
+            config[CONF_URL], config[CONF_USERNAME], config[CONF_PASSWORD]
+        )
+        if len(ncm.data) < 1:
+            _LOGGER.error(
+                "No data returned from Nextcloud API. Check your user/password."
+            )
+
+        hass.data[DOMAIN] = ncm.data
+
+        def nextcloud_update(event_time):
+            """Update data from nextcloud api."""
+            ncm.update()
+            hass.data[DOMAIN] = ncm.data
+
+        # Update sensors on time interval
+        track_time_interval(hass, nextcloud_update, config[CONF_SCAN_INTERVAL])
+
+        # Create list of sensors based on available nextcloud api data
+        sensors = []
+        for name, dict_path in NEXTCLOUD_ITEMS_EXPECTED.items():
+
+            try:
+                sensors.append(NextcloudSensor(name, dict_path))
+            except:  # noqa: E722 pylint: disable=bare-except
+                _LOGGER.warning(
+                    "%s sensor information was not provided by Nextcloud", name
+                )
+                continue
+
+        # Setup sensors
+        add_entities(sensors, True)
+    except:  # noqa: E722 pylint: disable=bare-except
+        raise PlatformNotReady
+
+
+class NextcloudSensor(Entity):
+    """Represents a Nextcloud sensor."""
+
+    def __init__(self, item, value_path) -> None:
```

Comment:
Remove the expected data type of the return value for all properties, methods and the constructor:
```suggestion
    def __init__(self, item, value_path)
```